cpp/src/binaryop/binaryop.cpp

```diff
@@ -571,26 +625,51 @@ std::unique_ptr<column> fixed_point_binary_operation(column_view const& lhs,
 std::unique_ptr<column> fixed_point_binary_operation(column_view const& lhs,
                                                      column_view const& rhs,
                                                      binary_operator op,
+                                                     thrust::optional<cudf::data_type> output_type,
                                                      rmm::cuda_stream_view stream,
                                                      rmm::mr::device_memory_resource* mr)
 {
   using namespace numeric;
 
   CUDF_EXPECTS(is_supported_fixed_point_binop(op), "Unsupported fixed_point binary operation");
-  CUDF_EXPECTS(lhs.type().id() == rhs.type().id(),
-               "Both columns must be of the same fixed_point type");
+  CUDF_EXPECTS(lhs.type().id() == rhs.type().id(), "Both columns must be of the same type");
+  CUDF_EXPECTS(op != binary_operator::TRUE_DIV || output_type.has_value(),
+               "TRUE_DIV requires result_type.");
 
-  auto const scale       = compute_scale_for_binop(op, lhs.type().scale(), rhs.type().scale());
-  auto const output_type = is_comparison_binop(op) ? data_type{type_id::BOOL8}  //
-                                                   : data_type{lhs.type().id(), scale};
-  auto out = make_fixed_width_column_for_output(lhs, rhs, op, output_type, stream, mr);
+  auto const scale = op == binary_operator::TRUE_DIV
+                       ? output_type.value().scale()
+                       : compute_scale_for_binop(op, lhs.type().scale(), rhs.type().scale());
+
+  auto const out_type = output_type.value_or(
+    is_comparison_binop(op) ? data_type{type_id::BOOL8} : data_type{lhs.type().id(), scale});
+
+  auto out = make_fixed_width_column_for_output(lhs, rhs, op, out_type, stream, mr);
 
   if (lhs.is_empty() or rhs.is_empty()) return out;
 
   auto out_view = out->mutable_view();
 
-  if (lhs.type().scale() != rhs.type().scale() && is_same_scale_necessary(op)) {
-    // Adjust columns so they have they same scale
+  if (op == binary_operator::TRUE_DIV) {
+    // Adjust columns so lhs has the scale needed to get desired output data_type (scale)
+    auto const diff         = lhs.type().scale() - rhs.type().scale() - scale;
+    auto const interim_op   = diff < 0 ? binary_operator::DIV : binary_operator::MUL;
+    auto const scalar_scale = scale_type{rhs.type().scale() + scale};
+    auto const result       = [&] {
+      if (lhs.type().id() == type_id::DECIMAL32) {
```

Comment:
It looks like there is significant repetition between the 3 different implementations (scalar column, column scalar, column column), but I'm not sure whether or not it can be eliminated cleanly? Suggest looking into it.

Reply:
Yea, i complete agree and am aware. I actually at one point started out with "one version" both the list of differences (`int32_t` vs `int64_t`, `decimal32` vs `decimal64`, `scalar` vs `column`, order of `lhs` & `rhs`, etc) started to make the resulting code messy. There is definitely a future "tech debt" job that will happen to consolidate the repetition that exists though.